fil_fungible_token/src/token/state.rs

```diff
@@ -1,25 +1,52 @@
-use anyhow::anyhow;
-use anyhow::bail;
-use anyhow::Result;
 use cid::multihash::Code;
 use cid::Cid;
-
 use fvm_ipld_blockstore::Block;
 use fvm_ipld_blockstore::Blockstore as IpldStore;
 use fvm_ipld_encoding::tuple::*;
 use fvm_ipld_encoding::Cbor;
 use fvm_ipld_encoding::CborStore;
 use fvm_ipld_encoding::DAG_CBOR;
+use fvm_ipld_hamt::Error as HamtError;
 use fvm_ipld_hamt::Hamt;
 use fvm_shared::bigint::bigint_ser;
 use fvm_shared::bigint::bigint_ser::BigIntDe;
 use fvm_shared::bigint::Zero;
 use fvm_shared::econ::TokenAmount;
 use fvm_shared::ActorID;
 use num_traits::Signed;
+use thiserror::Error;
 
 const HAMT_BIT_WIDTH: u32 = 5;
 
+#[derive(Error, Debug)]
+pub enum StateError {
+    #[error("ipld hamt error: {0}")]
+    IpldHamt(#[from] HamtError),
+    #[error("missing state at cid: {0}")]
+    MissingState(Cid),
+    #[error("underlying serialization error: {0}")]
+    Serialization(String),
+    #[error(
+        "negative balance caused by subtracting {delta:?} from {owner:?}'s balance of {balance:?}"
```

Comment:
The `delta` here will be negative when subtracting, so this double negative will be confusing.

Instead: "negative balance caused by _adding_ -1000 to owner's balance of 200". Or "changing balance by xxx"